### PR TITLE
Create general table component, use for all tables, delete imageTable

### PR DESCRIPTION
**Pull request by matoval**
Created general table component and used it in images.js. Deleted imageTable.js. 
Ticket [#1108](https://issues.redhat.com/browse/THEEDGE-1108)

### CODE REVIEW

File: src/Routes/ImageManager/Images.js [hunk 1]
```diff
@@ -111,6 +130,45 @@ const Images = () => {
     });
   };
 
+  const columnNames = [
```
Comment (boaz0) [inline src/Routes/ImageManager/Images.js:133]: This one will be re-created every time the `Image` component is re-rendered. Move it out of `Image`. Am I right?

File: src/Routes/ImageManager/Images.js [hunk 2]
```diff
@@ -166,6 +224,45 @@ const Images = () => {
...
+    if (rowData?.imageStatus !== 'SUCCESS') {
+      actionsArray.push({
+        title: '',
+      });
+    }
+
+    return actionsArray;
+  };
+
+  const areActionsDisabled = (rowData) => rowData?.imageStatus !== 'SUCCESS';
```
Comment (boaz0) [inline src/Routes/ImageManager/Images.js:265]: same here.

File: src/Routes/ImageManager/Images.js [hunk 3]
```diff
@@ -166,6 +224,45 @@ const Images = () => {
       },
     ],
   };
+  const filterDep = Object.values(activeFilters);
+
+  const actionResolver = (rowData) => {
```
Comment (boaz0) [inline src/Routes/ImageManager/Images.js:229]: Same here.

File: src/Routes/ImageManager/Images.js [hunk 4]
```diff
@@ -166,6 +224,45 @@ const Images = () => {
       },
     ],
   };
+  const filterDep = Object.values(activeFilters);
```
Comment (boaz0) [inline src/Routes/ImageManager/Images.js:227]: use memo here?

File: src/Routes/ImageManager/Images.js [hunk 5]
```diff
@@ -111,6 +130,45 @@ const Images = () => {
...
+  const columnNames = [
+    { title: 'Name', type: 'name', sort: true },
+    { title: 'Version', type: 'version', sort: false },
+    { title: 'Distribution', type: 'distribution', sort: true },
+    { title: 'Type', type: 'image_type', sort: false },
+    { title: 'Created', type: 'created_at', sort: true },
+    { title: 'Status', type: 'status', sort: true },
+  ];
+
+  const createRows = (data) => {
```
Comment (boaz0) [inline src/Routes/ImageManager/Images.js:142]: same here.

**Review by boaz0 — COMMENTED** [review]
Few comments.

**Review by resoluteCoder — APPROVED** [review]
lgtm

**Comment by resoluteCoder** [comment]
Hey @boaz0 what's up! :smile: , where specifically would you say there are unnecessary renders, I saw your suggestions to move certain vars/fn out of the Image component. It looks like @matoval changed from that

Thanks for any tips you can provide!

**Pull request merged**